bamboo-storage/src/main/java/com/pushtorefresh/android/bamboostorage/BambooStorage.java

```diff
@@ -20,44 +20,136 @@
+     *
+     * @return implementation of Internal operations for BambooStorage
+     */
     @NonNull Internal internal();
 
+    /**
+     * Hides some internal operations for BambooStorage to make API of BambooStorage clean and easy to understand
+     */
     interface Internal {
 
         /**
-         * Execute a single SQL statement that is NOT a SELECT/INSERT/UPDATE/DELETE.
+         * Execute a single SQL statement that is NOT a SELECT/INSERT/UPDATE/DELETE on the database
+         *
          * @param rawQuery sql query
          */
         void execSql(@NonNull RawQuery rawQuery);
 
+        /**
+         * Executes raw query on the database and returns {@link android.database.Cursor} over the result set
+         *
+         * @param rawQuery sql query
+         * @return A Cursor object, which is positioned before the first entry. Note that Cursors are not synchronized, see the documentation for more details.
+         */
         @NonNull Cursor rawQuery(@NonNull RawQuery rawQuery);
 
+        /**
+         * Executes query on the database and returns {@link android.database.Cursor} over the result set
+         *
+         * @param query sql query
+         * @return A Cursor object, which is positioned before the first entry. Note that Cursors are not synchronized, see the documentation for more details.
+         */
         @NonNull Cursor query(@NonNull Query query);
 
+        /**
+         * Inserts a row into the database
+         *
+         * @param insertQuery   query
+         * @param contentValues map that contains the initial column values for the row. The keys should be the column names and the values the column values
+         * @return id of inserted row
+         */
         long insert(@NonNull InsertQuery insertQuery, @NonNull ContentValues contentValues);
 
+        /**
+         * Updates one or multiple rows in the database
+         *
+         * @param updateQuery   query
+         * @param contentValues a map from column names to new column values. null is a valid value that will be translated to NULL.
+         * @return the number of rows affected
+         */
         int update(@NonNull UpdateQuery updateQuery, @NonNull ContentValues contentValues);
 
+        /**
+         * Deletes one or multiple rows in the database
+         *
+         * @param deleteQuery query
+         * @return the number of rows deleted
+         */
         int delete(@NonNull DeleteQuery deleteQuery);
 
+        /**
+         * Notifies subscribers about change in set of tables
+         * One operation can affect multiple tables, to reduce number of notifications
+         * you can call this method once and provide set of tables that were changed
+         *
+         * @param affectedTables set of affected tables
+         */
         void notifyAboutChanges(@NonNull Set<String> affectedTables);
 
+        /**
+         * BambooStorage implementation could not provide support for transactions
```

Comment:
By default: `BSSQLiteDatabase` provides support for transactions, I was wanted to say that different implementation of `BambooStorage` could not provide support for transactions